packages/angular_devkit/build_angular/src/dev-server/index.ts

```diff
@@ -233,17 +233,21 @@ export function serveWebpackBrowser(
       logger.warn(tags.stripIndent`
         Warning: This is a simple server for use in testing or debugging Angular applications
         locally. It hasn't been reviewed for security issues.
-
-        Binding this server to an open connection can result in compromising your application or
-        computer. Using a different host than the one passed to the "--host" flag might result in
-        websocket connection issues. You might need to use "--disableHostCheck" if that's the
-        case.
       `);
+
+      if (!options.disableHostCheck) {
+        logger.warn(tags.stripIndent`
+            Binding this server to an open connection can result in compromising your application or
```

Comment:
The "Binding this server...` sentence should probably stay in the warning above since it is the triggering element of the above warning.